docs/DeveloperGuide.adoc

```diff
@@ -767,6 +781,104 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+
+=== Use case: Undo Command
+
+*MSS*
+
+1.  User requests to undo previous command
+2.  AddressBook shows previous command to be undone.
+3.  User confirms to undo command.
+4.  AddressBook undo command.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There is no previous command.
++
+Use case ends.
+
+* 3a. The previous command is undoable.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Clear all
+
+*MSS*
+
+1.  User request to clear AddressBook
+2.  AddressBook prompts user for confirmation
+3.  User confirm the deletion
+4.  AddressBook clear all contact in AdressBook
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. User cancels the action.
++
+Use case ends.
+
+[discrete]
+=== Use case: List all persons
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+
+[discrete]
+=== Use case: Show user guide
+
+*MSS*
+
+1.  User requests to show user guide
+2.  AddressBook shows user guide on a separate link:#form[form]
+3.  User closes the user guide
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User guide cannot be found.
```

Comment:
not sure if this is a valid extension; this means you missed adding the user guide